hack/e2e-dynamicfile.sh

```diff
@@ -134,21 +208,22 @@ function e2e_dynamicfile(){
   if [ ! -z "$OUT" ]
       then
           echo $OUT
-          unset AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY AWS_SESSION_TOKEN
-          aws iam delete-role --role-name ${AWS_TEST_ROLE}
           echo "end to end testing for dynamicfile mode succeeded"
-          exit 0
 
       else
           echo "testing failed"
           exit 1
   fi
+  unset AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY AWS_SESSION_TOKEN
+  aws iam delete-role --role-name ${AWS_TEST_ROLE}
```

Comment:
Why delete this role on every test? That will cause a race issue if you have 2 tests running simultaneously